lnwallet/channel.go

```diff
@@ -3482,14 +3448,15 @@ func newHtlcResolution(signer Signer, localChanCfg *channeldb.ChannelConfig,
 		return nil, err
 	}
 
-	delayTweak := SingleTweakBytes(commitPoint,
+	// TODO: Signing with the delay key is wrong for remote commitments
+	localDelayTweak := SingleTweakBytes(keyRing.commitPoint,
 		localChanCfg.DelayBasePoint)
 	return &OutgoingHtlcResolution{
 		Expiry:          htlc.RefundTimeout,
 		SignedTimeoutTx: timeoutTx,
 		SweepSignDesc: SignDescriptor{
 			PubKey:        localChanCfg.DelayBasePoint,
-			SingleTweak:   delayTweak,
+			SingleTweak:   localDelayTweak,
```

Comment:
Note that we use the delay key here as this `SignDescriptor` is reserved for sweeping the output from the _second_ level HTLC, not spending directly from the commitment transaction. We already have the pre-signed `timeoutTx` at this state, so we can broadcast it directly to the network.